third_party/blink/renderer/modules/cobalt/h5vcc_experiments/h_5_vcc_experiments.cc

```diff
@@ -0,0 +1,161 @@
+    } else if (param_name_and_value.second->IsLong()) {
+      param_value = std::to_string(param_name_and_value.second->GetAsLong());
+    } else if (param_name_and_value.second->GetAsBoolean()) {
+      param_value = "true";
+    } else if (!param_name_and_value.second->GetAsBoolean()) {
+      param_value = "false";
+    } else {
+      // TODO(b/407734134): Do we need UMA on top of rejecting the promise?
+      resolver->Reject();
+    }
+    feature_params.Set(param_name_and_value.first.Utf8(), param_value);
+  }
+  experiment_config_dict.Set(cobalt::kExperimentConfigFeatureParams,
+                             std::move(feature_params));
+
+  base::Value::List experiment_ids;
+  for (int exp_id : experiment_configuration->experimentIds()) {
+    experiment_ids.Append(exp_id);
+  }
+  experiment_config_dict.Set(cobalt::kExperimentConfigExpIds,
+                             std::move(experiment_ids));
+
+  remote_h5vcc_experiments_->SetExperimentState(
+      std::move(experiment_config_dict),
+      WTF::BindOnce(&H5vccExperiments::OnSetExperimentState,
+                    WrapPersistent(this), WrapPersistent(resolver)));
+
+  return promise;
+}
+
+void H5vccExperiments::OnSetExperimentState(ScriptPromiseResolver* resolver) {
+  // TODO: Reject when this fails.
+  resolver->Resolve();
+}
+
+String H5vccExperiments::getFeature(const String& feature_name) {
+  EnsureReceiverIsBound();
+  h5vcc_experiments::mojom::blink::OverrideState result;
+  remote_h5vcc_experiments_->GetFeature(feature_name, &result);
+  switch (result) {
+    case h5vcc_experiments::mojom::blink::OverrideState::OVERRIDE_USE_DEFAULT:
+      return "OVERRIDE_USE_DEFAULT";
+    case h5vcc_experiments::mojom::blink::OverrideState::
+        OVERRIDE_ENABLE_FEATURE:
+      return "OVERRIDE_ENABLE_FEATURE";
+    case h5vcc_experiments::mojom::blink::OverrideState::
+        OVERRIDE_DISABLE_FEATURE:
+      return "OVERRIDE_DISABLE_FEATURE";
+  }
+  NOTREACHED_NORETURN() << "Invalid feature OverrideState: " << result;
+}
+
+const String& H5vccExperiments::getFeatureParam(
+    const String& feature_param_name) {
+  EnsureReceiverIsBound();
+  remote_h5vcc_experiments_->GetFeatureParam(feature_param_name,
+                                             &feature_param_value_);
+  return feature_param_value_;
+}
+
+ScriptPromise H5vccExperiments::resetExperimentState(
+    ScriptState* script_state,
+    ExceptionState& exception_state) {
+  auto* resolver = MakeGarbageCollected<ScriptPromiseResolver>(
+      script_state, exception_state.GetContext());
+
+  EnsureReceiverIsBound();
+
+  remote_h5vcc_experiments_->ResetExperimentState(
+      WTF::BindOnce(&H5vccExperiments::OnResetExperimentState,
+                    WrapPersistent(this), WrapPersistent(resolver)));
+
+  return resolver->Promise();
+}
+
+void H5vccExperiments::OnResetExperimentState(ScriptPromiseResolver* resolver) {
+  // TODO: Reject when this fails.
```

Comment:
Not sure when/how we should reject the promise since ResetExperiment doesn't have a return value. @joeltine wdyt..?

Reply:
Unsure. @yell0wd0g for input. Miguel, "resetExperimentState()" JS API returns a promise, but it seems we don't have a failure state such that a reject is even possible. I'd _think_ we'd reject of the mojo request to the browser process itself failed.... where would we catch that?